docker/ubuntu2204_clang.docker

```diff
@@ -37,6 +39,7 @@ RUN useradd -ms /bin/bash casacore \
 USER casacore
 RUN mkdir /code/build
 WORKDIR /code/build
+ENV CXXFLAGS="${CXXFLAGS} -stdlib=libc++"
```

Comment:
Why use libc++ instead of libstdc++? I think it would be good if also the tests work with libstdc++. If a part of it does not, we could work around that by ifdef'ing that part.

Reply:
This is not strictly required anymore, so it could be removed again. However, this is because of the extra 'boost_system' requirement, which results in that a part of the tests no longer is compiled. If they are compiled, the version of libstdc++ seemed at least too old to use the filesystem things in the test (I couldn't quite figure out why, but I thought that not running a test on an older platform is fine-ish)._ Also, clang is really made to compile with libc++, plus having on test run with that is a nice extra compatibility check, hence I kept these changes. If you rather not I'm also happy to remove them again.